### PR TITLE
Feature/navbar

### PR DESCRIPTION
**Pull request by No3ml3**
## Description

Please include a summary of the change or which issue is fixed.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Area of change

- [x] Frontend
- [ ] Backend
- [ ] Other: .....

## General checklist:

- [x] My code follows the style guidelines of this project
- [x] I ran `bun run format`/`bun run lint` before commit
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation where needed
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] I have updated `develop` and merged to my branch before submitting pull request
- [x] My pull request generate no conflicts with `develop` branch
- [x] I requested code review from other team members

## Frontend checklist:

- [x] I followed guidelines for `HTML`/`JSX`/`Typescript`
- [x] My `Javascript` generate no new console errors
- [x] I tested my code cross browsers
- [x] My slice is pixel perfect for both desktop and mobile according to design
- [x] I conducted basic QA to assure all features are working
- [x] I tested responsive for mobile and tablet resolutions

## Backend checklist:

- [ ] I tested admin by manually adding content from zero
- [ ] I created easy to use admin experience which is self-explanatory
- [ ] I added description to admin fields in hard-to-understand areas
- [ ] I followed guidelines for naming `Typescript` variables
- [ ] I conducted basic QA to assure all features are working

### CODE REVIEW

File: packages/frontend/web/src/components/Navbar.tsx
Comment (Justicesvr) [inline packages/frontend/web/src/components/Navbar.tsx:1]: elle est trop bien notre navbar!!

File: packages/frontend/web/src/pages/Home.tsx
Comment (Justicesvr) [inline packages/frontend/web/src/pages/Home.tsx:1]: et oui les imports mal écris qui faisaient tout bugger

**Review by Justicesvr — DISMISSED** [review]

**Review by jaalazard — DISMISSED** [review]

File: packages/frontend/web/src/pages/Home.tsx [hunk 1]
```diff
@@ -135,16 +131,15 @@ const squares = [
       lg: 0,
     },
     component: (
-      <Link to='/virgin'>
-        <VirginPart />
+      <Link to='/cocktails'>
+        <CocktailsPart />
```
Comment (YDrogen) [inline packages/frontend/web/src/pages/Home.tsx:135]: Pourquoi ? Les `<Link>` sont des balises `<a>`, pourquoi mettre un component dedans ?

File: packages/frontend/web/src/components/Navbar.tsx [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+  Home,
+  Martini,
+  Milk,
+  Plus,
+  UserCircle2,
+  Users,
+} from 'lucide-react';
+import { useEffect, useState } from 'react';
+import { useLocation } from 'react-router-dom';
+import { Link } from 'react-router-dom';
```
Comment (YDrogen) [inline packages/frontend/web/src/components/Navbar.tsx]: Ça tient sur une ligne

File: packages/frontend/web/src/components/Navbar.tsx [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+      case '/': {
+        setColor('#000000');
+        setHover('#000000');
+        break;
+      }
+      case '/shaker': {
+        setColor('#000000');
+        setHover('#000000');
+        break;
+      }
```
Comment (jaalazard) [inline packages/frontend/web/src/components/Navbar.tsx]: Pourquoi ?

Comment (YDrogen) [inline packages/frontend/web/src/components/Navbar.tsx]: C'est en double là, non ? On pourrait le sortir hors du `switch`

File: packages/frontend/web/src/components/Navbar.tsx [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+            <Milk
+              className={`peer h-7 w-7 cursor-pointer sm:my-auto sm:me-3 text-[${color}] hover:text-[${hover}]`}
+            />
+            <p
+              className={`text-[${color}] absolute max-h-0 opacity-0 transition-all peer-hover:max-h-full peer-hover:opacity-70 sm:right-[245px]`}
+            >
+              {'Virgin'}
+            </p>
+          </Link>
+        </div>
```
Comment (YDrogen) [inline packages/frontend/web/src/components/Navbar.tsx]: Y'a des boucles qui se perdent

File: packages/frontend/web/tailwind.config.js [hunk 5]
```diff
@@ -123,6 +123,21 @@ export default {
     'bg-profile-picture-red',
     'bg-profile-picture-pink',
     'bg-profile-picture-purple',
+    'really-dark-blue',
```
Comment (YDrogen) [inline packages/frontend/web/tailwind.config.js]: ?

**Review by YDrogen — CHANGES_REQUESTED** [review]

File: packages/frontend/web/src/pages/Home.tsx (hunk 1 above)
Comment (No3ml3) [reply packages/frontend/web/src/pages/Home.tsx:135]: tu voudrais que je fasse exactement avec ça ? que j'entoure les composants  d'una balise button ? (et accessoirmeent casse absolument tout le style :cry:) que je mette les links a un autre endroit ?

Comment (No3ml3) [reply packages/frontend/web/src/pages/Home.tsx:135]: en vrai j'ai replacer la div d'un de mes composant par button et ça n'a pas cassé tout le style est ce que ça te conviendrais ?

Comment (YDrogen) [reply packages/frontend/web/src/pages/Home.tsx:135]: Non c'est ok, j'avais juste pas compris ce code

File: packages/frontend/web/src/components/Navbar.tsx [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+import { Link, useLocation } from 'react-router-dom';
+
+type IconProps = {
+  className?: string;
+};
+
+type NavbarContent = {
+  name: string;
+  url: string;
+  lucid: React.ComponentType<IconProps>;
```
Comment (YDrogen) [inline packages/frontend/web/src/components/Navbar.tsx]: Il s'agit d'un component qui s'appelle `Icon`
Tu peux donc appeler la prop: `Icon`

**Review by jaalazard — DISMISSED** [review]

File: packages/frontend/web/src/components/Navbar.tsx
Comment (Justicesvr) [inline packages/frontend/web/src/components/Navbar.tsx:1]: plutot cool le switch dans un use effect

**Review by Justicesvr — DISMISSED** [review]

**Review by YDrogen — CHANGES_REQUESTED** [review]

**Review by Justicesvr — DISMISSED** [review]

**Review by jaalazard — DISMISSED** [review]

File: packages/frontend/web/src/components/Navbar.tsx (hunk 6 above)
Comment (YDrogen) [reply packages/frontend/web/src/components/Navbar.tsx]: Je parle de la **prop**, pas du type

**Review by Cardamced — APPROVED** [review]

**Review by jaalazard — APPROVED** [review]

**Review by YDrogen — APPROVED** [review]

**Pull request merged**